node/cn/api_backend_test.go

```diff
@@ -0,0 +1,544 @@
+}
+
+func TestCNAPIBackend_GetTxAndLookupInfo(t *testing.T) {
+	mockCtrl, mockBlockChain, _, api := newCNAPIBackend(t)
+	defer mockCtrl.Finish()
+
+	hash := hashes[0]
+
+	expectedTx := tx1
+	expectedBlockHash := hashes[1]
+	expectedBlockNum := uint64(111)
+	expectedIndex := uint64(999)
+
+	mockBlockChain.EXPECT().GetTxAndLookupInfo(hash).Times(1).Return(expectedTx, expectedBlockHash, expectedBlockNum, expectedIndex)
+	tx, blockHash, blockNumber, index := api.GetTxAndLookupInfo(hash)
+
+	assert.Equal(t, expectedTx, tx)
+	assert.Equal(t, expectedBlockHash, blockHash)
+	assert.Equal(t, expectedBlockNum, blockNumber)
+	assert.Equal(t, expectedIndex, index)
+}
+
+func TestCNAPIBackend_GetBlockReceipts(t *testing.T) {
+	mockCtrl, mockBlockChain, _, api := newCNAPIBackend(t)
+	defer mockCtrl.Finish()
+
+	hash := hashes[0]
+	expectedReceipts := types.Receipts{newReceipt(111), newReceipt(222)}
+
+	mockBlockChain.EXPECT().GetReceiptsByBlockHash(hash).Return(expectedReceipts).Times(1)
+
+	assert.Equal(t, expectedReceipts, api.GetBlockReceipts(context.Background(), hash))
+}
+
+func TestCNAPIBackend_GetLogs(t *testing.T) {
+	mockCtrl, mockBlockChain, _, api := newCNAPIBackend(t)
+	defer mockCtrl.Finish()
+
+	hash := hashes[0]
+	expectedLogs := [][]*types.Log{{{BlockNumber: 123}}, {{BlockNumber: 321}}}
+	mockBlockChain.EXPECT().GetLogsByHash(hash).Return(expectedLogs).Times(1)
+
+	logs, err := api.GetLogs(context.Background(), hash)
+	assert.Equal(t, expectedLogs, logs)
+	assert.NoError(t, err)
+}
+
+func TestCNAPIBackend_GetTd(t *testing.T) {
+	mockCtrl, mockBlockChain, _, api := newCNAPIBackend(t)
+	defer mockCtrl.Finish()
+
+	td := big.NewInt(123)
+	hash := hashes[0]
+	mockBlockChain.EXPECT().GetTdByHash(hash).Return(td).Times(1)
+
+	assert.Equal(t, td, api.GetTd(hash))
+}
+
+func TestCNAPIBackend_SubscribeEvents(t *testing.T) {
+	mockCtrl, mockBlockChain, _, api := newCNAPIBackend(t)
+	mockTxPool := mocks.NewMockTxPool(mockCtrl)
+	api.cn.txPool = mockTxPool
+	defer mockCtrl.Finish()
+
+	rmCh := make(chan<- blockchain.RemovedLogsEvent)
+	ceCh := make(chan<- blockchain.ChainEvent)
+	chCh := make(chan<- blockchain.ChainHeadEvent)
+	csCh := make(chan<- blockchain.ChainSideEvent)
+	leCh := make(chan<- []*types.Log)
+	txCh := make(chan<- blockchain.NewTxsEvent)
+
+	sub := mocks3.NewMockSubscription(mockCtrl)
+
+	mockBlockChain.EXPECT().SubscribeRemovedLogsEvent(rmCh).Return(sub).Times(1)
+	mockBlockChain.EXPECT().SubscribeChainEvent(ceCh).Return(sub).Times(1)
+	mockBlockChain.EXPECT().SubscribeChainHeadEvent(chCh).Return(sub).Times(1)
+	mockBlockChain.EXPECT().SubscribeChainSideEvent(csCh).Return(sub).Times(1)
+	mockBlockChain.EXPECT().SubscribeLogsEvent(leCh).Return(sub).Times(1)
+
+	mockTxPool.EXPECT().SubscribeNewTxsEvent(txCh).Return(sub).Times(1)
+
+	api.SubscribeRemovedLogsEvent(rmCh)
+	api.SubscribeChainEvent(ceCh)
+	api.SubscribeChainHeadEvent(chCh)
+	api.SubscribeChainSideEvent(csCh)
+	api.SubscribeLogsEvent(leCh)
+
+	api.SubscribeNewTxsEvent(txCh)
```

Comment:
Why don't you check the returned `sub` is equal with given `sub` in EXPECT routine?